### PR TITLE
Bump conmon-rs code to latest `main`

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind dependency-change

#### What this PR does / why we need it:
Update the latest conmon-rs golang client code.

#### Which issue(s) this PR fixes:

None
#### Special notes for your reviewer:
Requires https://github.com/containers/conmon-rs/pull/702 to make the CI pass again.
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
@haircommander PTAL

File: internal/oci/runtime_pod.go [hunk 1]
```diff
@@ -249,7 +253,7 @@ func (r *runtimePod) AttachContainer(ctx context.Context, c *Container, inputStr
 	)
 
 	if inputStream != nil {
-		stdin = &conmonClient.In{Reader: inputStream}
+		stdin = &conmonClient.In{ReadCloser: io.NopCloser(inputStream)}
```
Comment (haircommander) [inline internal/oci/runtime_pod.go:256]: shouldn't Close do something here?

Comment (saschagrunert) [reply internal/oci/runtime_pod.go:256]: Yes we could create our own type and close when we want to. We already close stdin on the client side and I'm not aware of any use case where we would close it in CRI-O. r`

**Comment by saschagrunert** [comment]
Hm, I suspect a logging related issue because it always fails when it tries to read the pod logs:
https://github.com/cri-o/cri-o/actions/runs/3065800947/jobs/4950349003

**Comment by saschagrunert** [comment]
@haircommander I'd say we merge this to unblock https://github.com/containers/conmon-rs/pull/749, the conmonrs-e2e tests are still expected to fail but we're getting closer.

File: internal/oci/runtime_pod.go [hunk 2]
```diff
@@ -171,6 +176,7 @@ func (r *runtimePod) ExecSyncContainer(ctx context.Context, c *Container, cmd []
 		Command:  cmd,
 		Timeout:  uint64(timeout),
 		Terminal: c.terminal,
+		Stdin:    c.stdin,
```
Comment (haircommander) [inline internal/oci/runtime_pod.go:179]: one nit: I don't think we need this, I opened https://github.com/containers/conmon-rs/pull/750

**Comment by saschagrunert** [comment]
Only one case failing: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/cri-o_cri-o/6220/pull-ci-cri-o-cri-o-main-ci-e2e-conmonrs/1572942729908850688

**Comment by saschagrunert** [comment]
@haircommander PTAL again
/override ci/prow/ci-e2e-conmonrs

**Pull request merged**